Improve waypoints and add image variant (#9480)
minetest/minetest: doc/client_lua_api.txt

* `offset`: offset in pixels from position.

### `waypoint`

> Displays distance to selected world position. Accepts offset.

Reviewer: ```suggestion
Displays distance to specified world position, and some text. Accepts offset.
```

* `name`: The name of the waypoint.
* `text`: Distance suffix. Can be blank.

> * `precision`: Waypoint precision. Defaults to 10 if not set (for compatibility reasons).

Reviewer: ```suggestion
* `precision`: Waypoint precision. Defaults to 10.
```

> * `precision`: Waypoint precision. Defaults to 10 if not set (for compatibility reasons).

Reviewer: Does this support fractional values? Document please

Author: No, it does not (because the used item field is an uint)

  If set to 0, distance is not shown. Shown value is `floor(distance*precision)/precision`.

>   Accordingly, there will be `n` steps behind the decimal dot.

Reviewer: I don't understand what this means. What did you mean by n? Did you mean:

```suggestion
  When the precision is an integer multiple of 10, there will `log_10(precision)` digits after the decimal point.
```

>   `precision = 1000`, for example, would show values like `0.999`.

Reviewer: ```suggestion
  `precision = 1000`, for example, will show 3 decimal places (eg: `0.999`).
```

* `number:` An integer containing the RGB value of the color used to draw the
  text.
* `world_pos`: World position of the waypoint.
* `alignment`: The alignment of the waypoint.

### `image_waypoint`

> Same as `image`, but does not accept a `position`; the position is instead determined by `world_pos`, the world position of the waypoint.

Reviewer: Please copy this from lua_api.txt

Author: Will do.
